packages/client-twitter/src/post.ts

```diff
@@ -113,9 +146,185 @@ export class TwitterPostClient {
         this.runtime = runtime;
     }
 
+    /**
+     * Convert a data URL or file path to a Buffer
+     * @param input The data URL string or file path
+     * @returns Buffer containing the image data
+     */
+    private async imageToBuffer(input: string): Promise<Buffer> {
+        // Check if it's a data URL
+        if (input.startsWith('data:')) {
+            const matches = input.match(/^data:([A-Za-z-+\/]+);base64,(.+)$/);
+            if (!matches || matches.length !== 3) {
+                throw new Error('Invalid data URL');
+            }
+            return Buffer.from(matches[2], 'base64');
+        }
+        
+        // Otherwise treat it as a file path
+        return fs.readFile(input);
+    }
+
     private async generateNewTweet() {
-        elizaLogger.log("Generating new tweet");
+        elizaLogger.log("Generating new tweet...");
+        const rawChance = this.runtime.getSetting("IMAGE_GEN_CHANCE") || "30";
+        const imageGenChancePercent = parseFloat(rawChance.replace(/[^0-9.]/g, '')) || 30;
+        elizaLogger.log(`Image generation chance set to ${imageGenChancePercent}%`);
+        
+        const shouldGenerateImage = Math.random() < (Math.max(0, Math.min(100, imageGenChancePercent)) / 100);
+        elizaLogger.log(`Will ${shouldGenerateImage ? '' : 'not '}generate image for this tweet`);
```

Comment:
As [above](https://github.com/ai16z/eliza/pull/629/files#r1863978464), better to just use the already parsed settings here.